sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ordering.scala

```diff
@@ -19,18 +19,24 @@ package org.apache.spark.sql.catalyst.expressions
 
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.BindReferences.bindReferences
+import org.apache.spark.sql.catalyst.expressions.codegen.GenerateOrdering
 import org.apache.spark.sql.types._
 
 
+/**
+ * A base class for generated/interpreted row ordering.
+ */
+abstract class BaseOrdering extends Ordering[InternalRow]
```

Comment:
hmm, so this is basically a type alias. Is it really useful?

Reply:
Honestly, not sure. It seems this base class has been implemented for making it easily accessible from generated code? https://github.com/apache/spark/blob/d555f8fcc964e6e81f2d849de62d06877dca70c5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateOrdering.scala#L158

Reply:
ah i see, to work around type erasure

Reply:
Ah, I see.